azkaban-common/src/main/java/azkaban/project/Project.java

```diff
@@ -39,6 +39,7 @@
   private int version = -1;
   private long createTimestamp;
   private long lastModifiedTimestamp;
+  private String currentUploaderID;
```

Comment:
I share with @suvodeep-pyne's concern that this information doesn't seem to fit well in the logic definition of a project. 

Could you describe your new GenericValidator interface better? Maybe in the issue?